src/main/java/com/linecorp/armeria/client/Clients.java

```diff
@@ -83,6 +157,33 @@
     }
 
     /**
+     * Creates a new client that connects to the static server group specified by {@code serverGroup}
+     * using an alternative {@link RemoteInvokerFactory}.For example,
+     * <p>
+     * <pre>{@code
+     * StaticServerGroup serverGroup = new StaticServerGroup();
+     * serverGroup.addHost("127.0.0.1", 1234, 2).addHost("127.0.0.1", 2345, 4)
+     *     .strategy(StaticServerGroup.SelectStrategyType.ROUND_ROBIN).registry("myGroup");
+     *
+     * Clients.newClient("tbinary+http://myGroup/...", ...,"myGroup",...);
+     * }</pre>
+     *
+     * @param remoteInvokerFactory an alternative {@link RemoteInvokerFactory}
+     * @param uri                  the URI of the servers endpoint
+     * @param interfaceClass       the type of the new client
+     * @param serverGroup          the name of the server group
+     * @param options              the {@link ClientOptions}
+     */
+    public static <T> T newClient(RemoteInvokerFactory remoteInvokerFactory, String uri, String serverGroup,
+                                  Class<T> interfaceClass, ClientOptions options) {
+        Node node = ServerGroupRegistryHub.selectNode(serverGroup);
+        uri = uri.replaceFirst(serverGroup, node.address());
```

Comment:
A user will have to call `newClient()` every time when he or she wants to choose the next node, which is not correct. The expected behavior is getting a client only once and then the node is chosen when an invocation is made:

```java
MyService.Iface client = Clients.newClient(...);
client.foo(); // chooses server A
client.bar(); // chooses server B
```